app/Model/AuthKey.php

```diff
@@ -162,12 +175,27 @@ public function getAuthUserByAuthKey($authkey, $includeExpired = false)
 
         $possibleAuthkeys = $this->find('all', [
             'recursive' => -1,
-            'fields' => ['id', 'authkey', 'user_id', 'expiration', 'allowed_ips', 'read_only'],
+            'fields' => ['id', 'authkey', 'user_id', 'expiration', 'allowed_ips', 'read_only', 'unique_ips'],
             'conditions' => $conditions,
         ]);
         $passwordHasher = $this->getHasher();
         foreach ($possibleAuthkeys as $possibleAuthkey) {
-            if ($passwordHasher->check($authkey, $possibleAuthkey['AuthKey']['authkey'])) {
+            if ($passwordHasher->check($authkey, $possibleAuthkey['AuthKey']['authkey'])) {  // valid authkey
+                // store IP in db if not there yet
+                $remote_ip = $this->_remoteIp();
+                $update_db_ip = true;
+                if (is_array($possibleAuthkey['AuthKey']['unique_ips']) && in_array($remote_ip, $possibleAuthkey['AuthKey']['unique_ips'])) {
```

Comment:
The check
```php
is_array($possibleAuthkey['AuthKey']['unique_ips'])
```
in combination with this in the `else` branch is a bit weird to me.
```php
$possibleAuthkey['AuthKey']['unique_ips'][] = $remote_ip;
```
Why not do directly do ?
```php
$possibleAuthkey['AuthKey']['unique_ips'] = [$remote_ip];
```

Reply:
I removed the `is_array()` , that's indeed not needed (anymore). Was needed before I did the current `afterFind()` who initiates the array in any case.

For the  else`: I corrected the comment "first time **this** IP is seen" 
It's to append the new IP to the array, as the array could have contained other IPs from previous connections.